### ISSUE TITLE
Song's Chord

### ISSUE THREAD

**Issue by eliasalbro95** [thomasdilts/churchsongs#40]
Song's chord is depending on 2 values (key and scale), and in our filter parameters we only depend on the chord's text(for example: 'Cm' or 'G'), so when I store the selected chords from the user I can't pass them to the API, because the API needs (key and scale).

For example: 
user selection: Cm and G

what I need to do in the current situation is:
```
{
key: "C, G"
scale: "Minor, Major"
}
```
which it's not valid, because it works only for 1 chord.

Do you have any idea to fix this?

**Comment by thomasdilts** [comment]
Yes. You are more or less correct. Just a comment though, when you are searching for both Minor and Major then just leave out the scale search parameter. If you want only C and G then you need 2 searches. One search with the filters:

"filters":[{"field":"keynote", "like":"C"}]

and then another search with:
"filters":[{"field":"keynote", "like":"G"}]

Also, something that is perhaps a bug but I call it a functionality is that if you search for "C" you get both "C" and "C#". But if you search for "C#" then you get only "C#". This is because I use a function called "like" and the search for "C" becomes in reality a search for "%C%" which means anything containing a "C".

Further search Explanation:
This filtering system is built entirely on "AND" functionality.  That is that each filter is "ANDed" with the next filter which means that your results will be always less and less each time you add a new filter.

It seems you want also an "OR" functionality where you can search for "C or G".  As you have noticed, this does not exist. Eventually we could add this but only for the keynote field because adding both OR and AND for all filters would be a nightmare problem. For now I recommend just 2 searches instead of one.

**Comment by eliasalbro95** [comment]
1- The question: when I leave the scale parameter empty (for example I wanted to search for Cm 'OR' G) this means that the C Major(C) and G Minor (Gm) are included in the search?
if yes then the search is not what the client expects, and not what we want the filter to do

2- Exactly we need it OR functionality.

3- Yes, it might work with 2 API calls, but calling an API for each selected keynote is a nightmare, it may take a long time.

**My suggestion:**
I made a local variable added to the Song Get API body called Chord, which decides what is the chord depending on the keynote and scale variables, (Chord =keynote: C + scale: Minor => Cm), I don't know if adding such a field to the table would make it easier of might fix the 1st problem we have, and might help with the others.
